### PR TITLE
Rename Drop* to Fallback*

### PR DESCRIPTION
**Pull request by arnecls**
Implementing issue #99
I decided on using "Fallback" instead of Retry as it is shows the intend more clearly.
Filters are a special case as "fallback" is not a good match here. As of this we have a special naming here.

- FallbackStream is used instead of DropStream for producers
- FilteredStream is used instead of DropStream for filters
- DropMessage renamed to RouteOriginal
- Adjusted all other occurrences to match the new naming pattern
- The \_DROPPED\_ stream has been removed, by default messages are discarded
- Drop stream metrics have been removed as #86 will make this metric obsolete

### CODE REVIEW

File: core/messagehelper_test.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func TestDropMessage(t *testing.T) {
 
 	msg := NewMessage(nil, []byte("foo"), 0, mockRouter.StreamID())
 
-	err := DropMessage(msg)
+	err := RouteOriginal(msg, msg.GetRouter())
```
Comment (msiebeneicher) [inline core/messagehelper_test.go]: This unit test should move to the RouteOriginal() class tests

File: core/messagehelper_test.go [hunk 2]
```diff
@@ -111,7 +111,7 @@ func TestDropMessageByRouter(t *testing.T) {
 	// create message and test
 	msg := NewMessage(nil, []byte("foo"), 0, mockRouterA.StreamID())
 
-	err := DropMessageByRouter(msg, &mockRouterB)
+	err := RouteOriginal(msg, &mockRouterB)
```
Comment (msiebeneicher) [inline core/messagehelper_test.go]: same here

File: core/bufferedproducer.go [hunk 3]
```diff
@@ -108,30 +108,32 @@ func (prod *BufferedProducer) Enqueue(msg *Message, timeout *time.Duration) {
...
 			prod.Log.Error.Print("Producer: ", prod.id, "State: ", prod.GetState(), ", Router: ", StreamRegistry.GetStreamName(msg.StreamID()))
-			prod.Drop(msg)
+			prod.TryFallback(msg)
 		}
 	}()
 
 	// Run modulators and drop message if necessary
 	result := prod.Modulate(msg)
 	switch result {
-	case ModulateResultDrop:
```
Comment (msiebeneicher) [inline core/bufferedproducer.go]: no ModulateResultFallback handling instead?

Comment (arnecls) [inline core/bufferedproducer.go]: But it's there ..? It just moved below discard

Comment (msiebeneicher) [inline core/bufferedproducer.go]: sry - did not saw it correctly in the unified view - ignore the comment

**Pull request merged**